sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java

```diff
@@ -235,7 +235,9 @@ public void registerCoderForType(TypeDescriptor<?> type, Coder<?> coder) {
    *
    * @throws CannotProvideCoderException if a {@link Coder} cannot be provided
    */
+  // CHECKSTYLE.OFF: MissingDeprecated
   @Deprecated
+  // CHECKSTYLE.ON: MissingDeprecated
```

Comment:
Yea, I mean `getCoderFromTypeDescriptor`. Basically we do need at least one (and probably exactly one) public method where you can pass in a context of type-to-coder bindings. And the rest should be directed to use that.